tests/validation/cattlevalidationtest/core/test_container.py

```diff
@@ -117,30 +117,26 @@ def test_linking(client, test_name, managed_network):
     delete_all(client, [link_client, link_server, link_server2])
 
 
-def create_container(admin_client, net):
-    cmd = '/bin/bash -c "sleep 5; ip addr show eth0"'
-    c = admin_client.create_container(imageUuid=TEST_IMAGE_UUID,
-                                      networkIds=[net.id],
-                                      command=cmd)
-    c = admin_client.wait_success(c)
-    logs = c.logs()
-
-    return logs, c, c.primaryIpAddress
-
-
-def cleanup(container):
-        try:
-            container.stop(remove=True, timeout=0)
-        except:
-            # We've tried our best!
-            pass
+def test_ip_inject(client, managed_network, test_name):
+    cleanup_items = []
+    try:
```

Comment:
@cjellick Craig, question about ip allocation. Who is responsible for handing out the free ip address - python-agent or cattle server? And if its implemented on both sides with python-agent doing it for native containers, and cattle server for rancher ones, how do we do cross check for ip conflicts?